public/video-ui/src/actions/VideoActions/saveVideo.js

```diff
@@ -45,24 +45,29 @@ export function saveVideo(video) {
   return dispatch => {
     dispatch(requestVideoSave(video));
     return VideosApi.saveVideo(video.id, video)
-    .then(res => {
-      dispatch(receiveVideoSave(res))
-      return VideosApi.getVideoUsages(video.id)
-    })
-    .then(usages => {
-      dispatch(receiveVideoUsages(usages));
-      return VideosApi.updateCanonicalPages(
-        video,
-        usages,
-        'preview'
-      )
-      .then(() => dispatch(receiveVideoPageUpdate(video.title)))
-    })
-    .catch(error => {
-      const defaultError = 'Could not save video';
-      const conflictError = 'Could not save video as another user (or tab) is currently editing this video';
-      const message = error.status === 409 ? conflictError : defaultError;
-      dispatch(errorVideoSave(error, message));
-    });
+      .then(res => {
+        dispatch(receiveVideoSave(res));
+        return VideosApi.getVideoUsages(video.id)
+          .then(usages => {
+            dispatch(receiveVideoUsages(usages));
+            return VideosApi.updateCanonicalPages(
+              video,
+              usages,
+              'preview'
+            );
+          })
+          .then(() => dispatch(receiveVideoPageUpdate(video.title)))
+          .catch(error => {
+            const message = 'Could not update canonical page';
+            dispatch(errorVideoSave(error, message));
+          });
+      })
+      .catch(error => {
+        const defaultError = 'Could not save video';
+        const conflictError = 'Could not save video as another user (or tab) is currently editing this video';
+        const message = error.status === 409 ? conflictError : defaultError;
+        dispatch(errorVideoSave(error, message));
+        throw error;
```

Comment:
Why does this function re-throw the error? The callsites catch and ignore the errors (e.g. [here](https://github.com/guardian/media-atom-maker/blob/e3ce35ab74fdfd21f8ff464ef4d4066e592389bd/public/video-ui/src/pages/Video/index.jsx#L288-L289)), so it seems like it would involve less code to ignore the error here (and have the same effect).

I guess maybe there are callsites (maybe [this](https://github.com/guardian/media-atom-maker/blob/e3ce35ab74fdfd21f8ff464ef4d4066e592389bd/public/video-ui/src/components/VideoImages/VideoImages.jsx#L20)?) which don’t ignore the errors?

Reply:
The intention is to reject the promise so that the caller can handle the rejected promise appropriately. If we just catch the error the promise will resolve successfully.

The links you have provided don't work for me, could you possible update them?